accepted/2020/serializer/WriteableDomAndDynamic.md

```diff
@@ -0,0 +1,584 @@
+The `JsonSerializerOptions` instance needs to be specified in order for any run-time added custom converters to be found and to use `PropertyNameCaseInsensitive` for property lookup with `JsonObject`.
+
+It is cumbersome and error-prone to specify the options instance for every new element or property:
+```cs
+    var jObject = new JsonObject(options)
+    {
+        ["MyString"] = new JsonValue("Hello!", options),
+        ["MyBoolean"] = new JsonValue(false, options),
+        ["MyArray"] = new JsonArray(options)
+        {
+            new JsonValue(2, options),
+            new JsonValue(3, options),
+            new JsonValue(42, options)
+        },
+    }
+```
+so it is possible to omit the options instance for non-root members. When a property or element is added, the options instance is set to the parent's instance:
+```cs
+    var jObject = new JsonObject(options) // options still needed at root level
+    {
+        ["MyString"] = new JsonValue("Hello!"),
+        ["MyBoolean"] = new JsonValue(false),
+        ["MyArray"] = new JsonArray()
+        {
+            new JsonValue(2),
+            new JsonValue(3),
+            new JsonValue(42)
+        },
+    }
+```
+
+It is required that all option instances are the same across a given `JsonNode` tree.
+
+### Using `System.Object` for values
+`JsonNode` has indexers that allow `JsonObject` properties and `JsonArray` elements to be specified. This allows a terse and more performant mode:
+```cs
+    var jObject = new JsonObject(options)
+    {
+        ["Child"]["Array"][0]["Message"] = new JsonValue("Hello!");
+        ["Child"]["Array"][1]["Message"] = new JsonValue("Hello!!");
+    }
+```
+
+If `System.Object` was used instead of `JsonNode`, or the indexers were not exposed on `JsonNode`, the above terse syntax wouldn't work and a more verbose syntax would be necessary:
+```cs
+    object jObject = new JsonObject(options)
+    {
+        ((JsonObject)((JsonArray)((JsonObject)jObject["Child"])["Array"])[0])["Message"] = new JsonValue("Hello!");
+        ((JsonObject)((JsonArray)((JsonObject)jObject["Child"])["Array"])[1])["Message"] = new JsonValue("Hello!!");
+    }
+```
+
+During serialization, the appropriate converter is used to serialize either the primitive value or the underlying value held by a `JsonNode`.
+
+It is also possible to take a given CLR type and assign it to `Value`:
+``` cs
+    JsonArray jArray = ...
+    jArray.Value = new int[] {0, 1}; // raw array used instead of JsonValue
+    string json = jArray.Serialize();
+```
+
+## Changing the deserialization of System.Object
+The serializer has always deserialized JSON that maps to a `System.Object` as a `JsonElement`. This will remain the default behavior going forward, however there will be a way to specify `JsonNode` instead:
+
+```cs
+namespace System.Text.Json
+{
+    public partial class JsonSerializerOptions
+    {
+        public bool UseJsonNodeForUnknownTypes { get; set;}
```

Comment:
It seems we want this to be implicitly set to true when `EnableDynamicTypes()` is called.